### DISCUSSION TITLE
The :valid pseudo class is not working in my input

### DISCUSSION THREAD

**Discussion by TIEverton** [tailwindlabs/tailwindcss#12043] [Help] [answered]
I'm trying to leave the background color of my input as red, but when I type something in the input the pseudo class doesn't work and nothing happens.

```html
<input
  className="valid:bg-slate-950 text-sm bg-transparent h-full w-full outline-none pl-2"
/>
```

![image](https://github.com/tailwindlabs/tailwindcss/assets/44790240/a3bc7377-d719-4acf-bd6d-7cdb01d879d8)
![image](https://github.com/tailwindlabs/tailwindcss/assets/44790240/c8803d9f-7137-4793-b708-cb8df396b60d)

**Comment by wongjn** [comment]
`valid:` seems to be working fine here – the input seems to have a background color of `slate.950` from the `valid:bg-slate-950` class.

Could you clarify what you mean by "leave the background color of my input as red"?

**Reply by TIEverton** [reply]
When I run my project, even though the input is not filled in, it turns to the color bg-slate-950, which was only supposed to happen when I type something in the input.

**Reply by wongjn** [reply]
I think you are misunderstanding the concept of `valid:`. `valid:` is equivalent to the `:valid` CSS pseudo-class. The `<input>` element given in your example code would match `:valid` filled in or not, thus the `bg-slate-950` should always apply to the element, whether it is filled in or not.

**Reply by TIEverton** [reply]
And how could I put a background color only if the input is filled?

**Reply by wongjn** [reply, marked as answer]
You could consider applying the `required` attribute to the `<input>` element: https://play.tailwindcss.com/yQD3BQYZ9y
